qcodes/dataset/database_extract_runs.py

```diff
@@ -116,43 +114,6 @@ def extract_runs_into_db(source_db_path: str,
         target_conn.close()
 
 
-def _create_exp_if_needed(target_conn: ConnectionPlus,
```

Comment:
hmm, i think the reason this function is here is becasue it's private wrt the logic of extracting runs to another db. But i agree that it belongs to the experiment module, and it can be used to create experiments in the DB **with all their attributes provided** (starttime/endtime/fmtstr/etc), but not by experimentalists/researchers - they should still use the function that only takes exp_name and sample_name.

Reply:
Yes no one should use this method outside qcodes which is why it is still private. I wish python had stronger conversions about module/class/submodule private but perhaps we should just agree on our own? The intention with making this private is that is should not be used outside qcodes. The issue with leaving this in database_extract_run is that it very easily results in code duplication e.g. other modules that need to do something similar will implement the same functionality. 

I would prefer to get rid of it completely. I am not even sure comparing against these attributes even always makes sense. E.g. format_string is probably very rarely changed in practice. End_time could easily change between two different db files for the same experiment (if after moving the first dataset one db file has the experiment marked completed exporting a new dataset from and to the same db files will duplicate the experiment in the second one )